app/src/androidTest/java/com/github/h3lp3rs/h3lp/firstaid/AedActivityTest.kt

```diff
@@ -37,25 +48,38 @@ class AedActivityTest {
     }
     @Test
     fun tutorialVideoIsDisplayed(){
-        checkIfDisplayed(R.id.aedVideo)
+        launchAndDo {
+            checkIfDisplayed(R.id.aedVideo)
+        }
     }
 
     @Test
     fun tutorialDescriptionIsDisplayed(){
-        checkIfDisplayed(R.id.aedTutorialStep1)
-        checkIfDisplayed(R.id.aedTutorialStep2)
-        checkIfDisplayed(R.id.aedTutorialStep3)
+        launchAndDo {
+            checkIfDisplayed(R.id.aedTutorialStep1)
+            checkIfDisplayed(R.id.aedTutorialStep2)
+            checkIfDisplayed(R.id.aedTutorialStep3)
+        }
     }
 
     @Test
-    fun backButtonWorks(){
-        init()
-        onView(withId(R.id.aed_back_button)).perform(scrollTo(),ViewActions.click())
-        Intents.intended(
-            Matchers.allOf(
-                IntentMatchers.hasComponent(FirstAidActivity::class.java.name)
+    fun backButtonWorks() {
+        launchAndDo {
+            onView(withId(R.id.aed_back_button)).perform(scrollTo(), ViewActions.click())
+            Intents.intended(
+                Matchers.allOf(
+                    IntentMatchers.hasComponent(FirstAidActivity::class.java.name)
+                )
             )
-        )
-        release()
+        }
+    }
+
+
+    private fun launchAndDo(action: () -> Unit) {
```

Comment:
I'm down with this idea but there might be a problem, to define launchAndDo, we need a launch method. I tried to define an abstract launch method that returns an ActivityScenario<T> with T being a generic type parameter for the H3lpAppTest class. The problem stems from the fact that this launch method has to be abstract but now some ActivityTests need to override launch but don't even use it. I could put launchAndDo and launch into a separate test utils abstract class / interface but this isn't really pretty.
I'm interested by any suggestions

Reply:
### but now some ActivityTests need to override launch but don't even use it
I agree, I think the benefits outweigh the problems though. Also, this can be solved by defining an open function and making the override optional. This way, if one calls the open function without overriding it, it simply throws a `NotImplementedException`. 
What do you think?

Reply:
I'm fine with this option, I'll implement it